From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hazerdous <dransom96@gmail.com>
Date: Wed, 22 May 2019 12:07:20 -0400
Subject: [PATCH] Monumenta - Make NBT List comparisons ignore order


diff --git a/src/main/java/net/minecraft/nbt/NBTTagList.java b/src/main/java/net/minecraft/nbt/NBTTagList.java
index ce6572df63c4e7341708aee60330fb214a3fe416..7b3f92563251cd3f56048ebf2f4b6beb46195321 100644
--- a/src/main/java/net/minecraft/nbt/NBTTagList.java
+++ b/src/main/java/net/minecraft/nbt/NBTTagList.java
@@ -291,7 +291,7 @@ public class NBTTagList extends NBTList<NBTBase> {
     }
 
     public boolean equals(Object object) {
-        return this == object ? true : object instanceof NBTTagList && Objects.equals(this.list, ((NBTTagList) object).list);
+        return this == object ? true : object instanceof NBTTagList && this.list.size() == ((NBTTagList) object).list.size() && this.list.containsAll(((NBTTagList) object).list);
     }
 
     public int hashCode() {
