From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Thu, 6 Feb 2020 01:34:38 -0500
Subject: [PATCH] Monumenta - Refuse to start shard if scoreboard fails to load

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/world/level/storage/WorldPersistentData.java b/src/main/java/net/minecraft/world/level/storage/WorldPersistentData.java
index d61960d80599dc5e7b70cc990e4b0b174eb6e34e..5abfd437ce32af0421a478a2588a0e9cd03a2510 100644
--- a/src/main/java/net/minecraft/world/level/storage/WorldPersistentData.java
+++ b/src/main/java/net/minecraft/world/level/storage/WorldPersistentData.java
@@ -77,6 +77,7 @@ public class WorldPersistentData {
             }
         } catch (Exception exception) {
             WorldPersistentData.LOGGER.error("Error loading saved data: {}", s, exception);
+            System.exit(-9001);
         }
 
         return null;
