From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Fri, 13 Mar 2020 17:33:42 -0400
Subject: [PATCH] Monumenta - Make armor unbreaking work the same as tools

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index 366ceb65d40c685117d1c79a933864ab91d8aa11..e8fee121df4c6a59a894e4689aaaaab0f42cfa3f 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -494,6 +494,12 @@ public final class ItemStack {
                 j = EnchantmentManager.getEnchantmentLevel(Enchantments.DURABILITY, this);
                 int k = 0;
 
+                /* MONUMENTA:
+                 * Reduce giant spikes of gear damage to make high-damage mobs less destructive to gear
+                 * https://www.wolframalpha.com/input/?i=plot+min%28x%2F8%2B2%2Cx%2F4%29+and+min%28sqrt%28x%29%2Cx%2F4%29+and+x%2F4%2C+x%3D1..40
+                 */
+                i = (int)Math.min(i, Math.sqrt(i * 4));
+
                 for (int l = 0; j > 0 && l < i; ++l) {
                     if (EnchantmentDurability.a(this, j, random)) {
                         ++k;
diff --git a/src/main/java/net/minecraft/world/item/enchantment/EnchantmentDurability.java b/src/main/java/net/minecraft/world/item/enchantment/EnchantmentDurability.java
index 06ce97fe5b5ddd9aba0c10ea082af9899ad52150..5bc95cd773a3906cbc07375602cee1628ffcfa66 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/EnchantmentDurability.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/EnchantmentDurability.java
@@ -32,6 +32,7 @@ public class EnchantmentDurability extends Enchantment {
     }
 
     public static boolean a(ItemStack itemstack, int i, Random random) {
-        return itemstack.getItem() instanceof ItemArmor && random.nextFloat() < 0.6F ? false : random.nextInt(i + 1) > 0;
+        /* MONUMENTA: All items work like tools for unbreaking */
+        return random.nextInt(i + 1) > 0;
     }
 }
