From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Byron Marohn <combustible@live.com>
Date: Wed, 29 Apr 2020 21:17:15 +0000
Subject: [PATCH] Monumenta - Reset last player hurt time on taking any damage

Signed-off-by: Byron Marohn <combustible@live.com>

diff --git a/src/main/java/net/minecraft/world/entity/EntityLiving.java b/src/main/java/net/minecraft/world/entity/EntityLiving.java
index b7b07b652b29e6f84f87fc92add99ce68f8bbd09..a25df6ed807cbc4c572008dfd9d51430d74dd8f8 100644
--- a/src/main/java/net/minecraft/world/entity/EntityLiving.java
+++ b/src/main/java/net/minecraft/world/entity/EntityLiving.java
@@ -1280,6 +1280,10 @@ public abstract class EntityLiving extends Entity {
             this.ap = 0.0F;
             Entity entity1 = damagesource.getEntity();
 
+            if (this.killer != null) {
+                this.lastDamageByPlayerTime = 100;
+            }
+
             if (entity1 != null) {
                 if (entity1 instanceof EntityLiving) {
                     this.setLastDamager((EntityLiving) entity1);
